templates/common/specutil/spec.go

```diff
@@ -102,3 +111,28 @@ func FormatAttrList(w io.Writer, attrList [][]string) {
 	}
 	tw.Flush()
 }
+
+// Load unmarshals the spec.yaml in the given directory.
+func Load(ctx context.Context, fs common.FS, templateDir, source string) (*spec.Spec, error) {
+	specPath := filepath.Join(templateDir, SpecFileName)
+	f, err := fs.Open(specPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("couldn't find spec.yaml in that directory, the provided template location %q might be incorrect", source)
+		}
+		return nil, fmt.Errorf("error opening template spec: Open(): %w", err)
+	}
+	defer f.Close()
+
+	specI, err := decode.DecodeValidateUpgrade(ctx, f, SpecFileName, decode.KindTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("error reading template spec file: %w", err)
+	}
+
+	spec, ok := specI.(*spec.Spec)
+	if !ok {
+		return nil, fmt.Errorf("internal error: spec file did not decode to spec.Spec")
```

Comment:
question, would changing `spec.Spec` to `v1beta1.Spec` provide more context? As this files is also `spec.go`, might cause confusion.

Reply:
I don't think we want that. We want to be able to change the spec version number in the future without updating this error message.

Reply:
Thanks!